frontend/src/metabase/common/components/Table/Table.tsx

```diff
@@ -1,69 +1,110 @@
-import React from "react";
+import type React from "react";
+import {
+  Fragment,
+  type PropsWithChildren,
+  useCallback,
+  useMemo,
+  useState,
+} from "react";
 
 import { color } from "metabase/lib/colors";
+import { useSelector } from "metabase/lib/redux";
+import { getLocale } from "metabase/setup/selectors";
 import { Box, Flex, Icon } from "metabase/ui";
 
-type BaseRow = Record<string, unknown> & { id: number | string };
+export type BaseRow = Record<string, any> & { id: number | string };
 
 type ColumnItem = {
   name: string;
   key: string;
+  sortable?: boolean;
 };
 
 export type TableProps<Row extends BaseRow> = {
+  /** An array of objects with name and key properties */
   columns: ColumnItem[];
+  /** An array of row objects, which at minimum need an id */
   rows: Row[];
+  /** A function that takes a row object and returns a <tr> element */
   rowRenderer: (row: Row) => React.ReactNode;
-  tableProps?: React.HTMLAttributes<HTMLTableElement>;
-};
+
+  /** A function that takes a row object and a column key and returns a value to use for sorting */
+  formatValueForSorting?: (row: Row, columnName: string) => any;
+  defaultSortColumn?: string;
+  defaultSortDirection?: "asc" | "desc";
+
+  /** A component to render if the table is empty */
+  ifEmpty?: React.ReactNode;
+} & React.HTMLAttributes<HTMLTableElement>;
 
 /**
  * A basic reusable table component that supports client-side sorting by a column
- *
- * @param columns     - an array of objects with name and key properties
- * @param rows        - an array of objects with keys that match the column keys
- * @param rowRenderer - a function that takes a row object and returns a <tr> element
- * @param tableProps  - additional props to pass to the <table> element
```

Comment:
I think the goal here was to make the documentation for the props visible in IDEs when you look up the documentation for the component. Strictly speaking, though, this isn't valid JSDoc (there is only one param, not multiple) and this was boggling my head and my IDE. `tableProps` also works differently than described in this comment: it's not a distinct prop. I've boldly gone ahead and converted this to a documentation pattern I'm familiar with, even though this means the comments are not visible when you look up the documentation for the component, which is a shame. I will look into whether it's possible to improve this situation.

Reply:
Perhaps we can use this pattern from JSDoc:
```
/**
 * A table component
 * @param {object} props
 * @property columns - An array of objects with name and key properties
 * @property rows - An array of row objects
 */
```

I've posted about this on StackOverflow: https://stackoverflow.com/questions/78627351/documenting-props-in-a-react-typescript-component/78627377#78627377 Still thinking about this. I think we need a good pattern we can begin to stick with, to establish a clearer "API" for each major component.

Reply:
I've now introduced that jsdoc pattern in the Table component.